shared/src/Datadog.Trace.ClrProfiler.Native/process_helper.cpp

```diff
@@ -57,26 +86,35 @@ bool ProcessHelper::RunProcess(const std::string& processPath,
     // Because the equal sign is a separator, it must not be used in the name of an environment variable.
 
     PROCESS_INFORMATION pi;
-    if (CreateProcess(
+    if (!CreateProcess(
         nullptr,                  // lpApplicationName
         commandLine.data(),       // lpCommandLine
         nullptr,                  // lpProcessAttributes
         nullptr,                  // lpThreadAttributes
-        FALSE,                    // bInheritHandles
+        TRUE,                     // bInheritHandles
         CREATE_NEW_PROCESS_GROUP, // dwCreationFlags (don't create as a child)
         nullptr,                  // lpEnvironment
         nullptr,                  // lpCurrentDirectory
         &si,                      // lpStartupInfo
         &pi))                     // lpProcessInformation
     {
-        // we don't need to wait for it to finish, so just free-up resources 
-        CloseHandle(pi.hProcess);
-        CloseHandle(pi.hThread);
-        return true;
-    }
+        Log::Warn("ProcessHelper::RunProcess: Error starting ", processPath);
+        CloseHandle(hStdinRead);
+        CloseHandle(hStdinWrite);
+        return false;
+    }    
+
+    // we don't need to wait for it to finish, so just free-up resources 
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+
+    DWORD written;
+    WriteFile(hStdinWrite, input.c_str(), input.length(), &written, nullptr);
+
+    CloseHandle(hStdinRead);
+    CloseHandle(hStdinWrite);
```

Comment:
So it's fine to do this _after_ we've closed the `pi.` handles right? Is there any specific reason to do it one way or the other?

Reply:
I don't think the order matters. I guess I could group the CloseHandle calls for clarity 🤔